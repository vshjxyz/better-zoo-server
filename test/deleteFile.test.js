import fs from 'fs'

let deleteFile

describe('deletes a file from filesystem', () => {
  beforeEach(() => {
    jest.spyOn(fs, 'unlink')
    deleteFile = require('../src/lib/deleteFile').default
  })

  it('resolves if file exists and no errors were thrown', async () => {
    const path = 'return value from deleteFile is ciao'
    fs.unlink.mockImplementationOnce((path, callback) => callback())
    await deleteFile(path)
    expect(fs.unlink).toHaveBeenCalled()
    expect(fs.unlink.mock.calls[0][0]).toEqual(path)
  })

  it('rejects if there is an error during unlink', async () => {
    const path = 'return value from deleteFile is ciao'
    const error = new Error('error')
    fs.unlink.mockImplementationOnce((path, callback) => callback(error))
    await expect(deleteFile(path)).rejects.toThrowErrorMatchingSnapshot()
  })

  it('resolves if there is an ENOENT error', async () => {
    const path = 'return value from deleteFile is ciao'
    const error = new Error('error')
    error.code = 'ENOENT'
    fs.unlink.mockImplementationOnce((path, callback) => callback(error))
    await deleteFile(path)
    expect(fs.unlink).toHaveBeenCalled()
  })
})
