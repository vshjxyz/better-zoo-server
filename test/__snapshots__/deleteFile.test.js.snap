// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deletes a file from filesystem rejects if there is an error during unlink 1`] = `"error"`;
